### ISSUE TITLE
ONNX export of Mask-RCNN doesn't output masks

### ISSUE THREAD

**Issue by JeroenDelcour** [pytorch/vision#3291]
## 🐛 Bug

I trained a Mask-RCNN model, exported it to ONNX, and tried to run it using onnxruntime. Sometimes, the model outputs boxes, labels, and scores as expected, but outputs 0 masks. I'm not entirely sure when or why this happens, but it's reproducible.

## To Reproduce

I used the default Mask-RCNN model from the model zoo and tested it on this image from the COCO database:

```bash
wget https://farm9.staticflickr.com/8289/7717265494_a6c6ac01bb_z.jpg
```
![](https://farm9.staticflickr.com/8289/7717265494_a6c6ac01bb_z.jpg)

```python
import torch
import torchvision
import numpy as np
import onnxruntime
from PIL import Image
torch.manual_seed(0)

model = torchvision.models.detection.maskrcnn_resnet50_fpn(pretrained=True, num_classes=91)
model.eval()

dummy_input = torch.rand(1, 3, 1333, 1333)

torch.onnx.export(model,
                  dummy_input,
                  "maskrcnn.onnx",
                  opset_version=11,
                  input_names=["input"],
                  dynamic_axes={"input": {0: "batch", 2: "height", 3: "width"}},
                  )
# load test image
image = Image.open("7717265494_a6c6ac01bb_z.jpg")
image = np.float32(image)
image = np.moveaxis(image, -1, 0)  # HWC to CHW
# normalize
mean_vec = np.array([0.485, 0.456, 0.406])
stddev_vec = np.array([0.229, 0.224, 0.225])
for i in range(image.shape[0]):
    image[i, :, :] = (image[i, :, :]/255 - mean_vec[i]) / stddev_vec[i]
# add batch dimension
image = image[np.newaxis, :]

# run model
with open("maskrcnn.onnx", "rb") as f:
    sess = onnxruntime.InferenceSession(f.read())
input_name = sess.get_inputs()[0].name
pred = sess.run(None, {input_name: image})
boxes, labels, scores, masks = pred

print(len(boxes), len(labels), len(scores), len(masks))
```

Outputs (besides lots of warnings): `18, 18, 18, 0`.

## Expected behavior

There should be a mask for every bounding box.

## Environment

```
PyTorch version: 1.7.1
Is debug build: False
CUDA used to build PyTorch: 10.2
ROCM used to build PyTorch: N/A

OS: Ubuntu 20.04.1 LTS (x86_64)
GCC version: (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0
Clang version: Could not collect
CMake version: version 3.16.3

Python version: 3.8 (64-bit runtime)
Is CUDA available: True
CUDA runtime version: Could not collect
GPU models and configuration: GPU 0: GeForce GTX 950M
Nvidia driver version: 450.102.04
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] numpy==1.19.5
[pip3] torch==1.7.1
[pip3] torchvision==0.8.2
[conda] Could not collect
```

 - PyTorch / torchvision Version (e.g., 1.0 / 0.4.0): 1.7.1 / 0.8.2
 - OS (e.g., Linux): Ubuntu
 - How you installed PyTorch / torchvision (`conda`, `pip`, source): pip
 - Build command you used (if compiling from source):
 - Python version: 3.8
 - CUDA/cuDNN version:
 - GPU models and configuration: GeForce GTX 950M
 - Any other relevant information:

## Additional context

It seems to depend on the model weights and the dummy input for exporting. Some exported models work fine, some models output 0 masks regardless of input, and some models output masks but there are fewer masks than boxes. The number of boxes, labels, and scores has so far always been equal.

I tried setting the outputs as dynamic axes, but that doesn't seem to have any effect. I also tried using an image from the training dataset as the dummy input, and it seems using dummy input images with more objects in them increases the maximum masks the exported model can output, but it's not very consistent.

cc @neginraoof

**Comment by nesterova-anna** [comment]
Hi, @JeroenDelcour! Did you solve this problem? I have same situation and don't know way to fix(

**Comment by JeroenDelcour** [comment]
> Hi, @JeroenDelcour! Did you solve this problem? I have same situation and don't know a way to fix(

No, unfortunately. My current workaround is to use an image with lots of detectable objects as a dummy input image. That yields a working onnx model that outputs enough masks for my use case, though still fewer than it's supposed to.

**Comment by QNester** [comment]
Any updates? We really need this feature :(

**Comment by JeroenDelcour** [comment]
I don't know what did it, but this appears to be fixed with the latest version of torchvision :)
